influent-java/src/main/java/influent/forward/NioForwardServer.java

```diff
@@ -58,18 +85,109 @@
                    final boolean keepAliveEnabled,
                    final boolean tcpNoDelayEnabled,
                    final int workerPoolSize) {
+    this(localAddress, callback, chunkSizeLimit, backlog,
+        sendBufferSize, receiveBufferSize, keepAliveEnabled, tcpNoDelayEnabled,
+        workerPoolSize, Protocol.TCP, TlsVersion.None);
+    context = null;
+  }
+
+  NioForwardServer(final SocketAddress localAddress,
+                   final ForwardCallback callback,
+                   final long chunkSizeLimit,
+                   final int backlog,
+                   final int sendBufferSize,
+                   final int receiveBufferSize,
+                   final boolean keepAliveEnabled,
+                   final boolean tcpNoDelayEnabled,
+                   final int workerPoolSize,
+                   final Protocol protocol,
+                   final TlsVersion tlsVersion) {
     bossEventLoop = NioEventLoop.open();
     workerEventLoopPool = NioEventLoopPool.open(workerPoolSize);
-    final Consumer<SocketChannel> channelFactory = socketChannel -> new NioForwardConnection(
-        socketChannel, workerEventLoopPool.next(), callback, chunkSizeLimit, sendBufferSize,
-        keepAliveEnabled, tcpNoDelayEnabled
-    );
-    new NioTcpAcceptor(
-        localAddress, bossEventLoop, channelFactory, backlog, receiveBufferSize
-    );
+    this.protocol = protocol;
+    this.tlsVersion = tlsVersion;
+    switch (protocol) {
+      case TCP:
+        context = null;
+        final Consumer<SocketChannel> tcpChannelFactory = socketChannel -> new NioForwardConnection(
+            socketChannel, workerEventLoopPool.next(), callback, chunkSizeLimit, sendBufferSize,
+            keepAliveEnabled, tcpNoDelayEnabled
+        );
+        new NioTcpAcceptor(
+            localAddress, bossEventLoop, tcpChannelFactory, backlog, receiveBufferSize
+        );
+        break;
+      case TLS:
+        try {
+          context = SSLContext.getInstance(tlsVersion.toString());
```

Comment:
It should be `SSLEngine.setEnabledProtocols` where protocols are chosen.
Using `SSLContext.getInstance`, non-strict protocols are enabled and `SSLEngine` seems not to extend `SSLContext's` configuration.
It looks good to always use `SSLContext.getInstance("TLS")`.
https://github.com/okumin/nio-ssl-test/blob/master/src/main/java/niossltest/ProtocolPrinter.java

And more, server should accept several versions for compatibility.
So it is desirable that users can pass multiple protocols via ForwardServer.Builder.